File: FinTrack/Models/Category.cs
namespace FinTrack.Models;

public class Category
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public string Name { get; set; } = null!;
    public CategoryType Type { get; set; }
    public Guid? ParentCategoryId { get; set; }
    public Category? ParentCategory { get; set; }
    public bool IsArchived { get; set; }
    public ICollection<Category> Children { get; set; } = [];
    public ICollection<Transaction> Transactions { get; set; } = [];
}

public enum CategoryType : short
{
    Income = 1,
    Expense = 2
}

File: FinTrack.Client/src/types/category.ts
export interface CreateCategoryRequest {
    name: string;
    type: CategoryType;
    parentCategoryId: string | null;
}

export interface UpdateCategoryRequest {
    name?: string;
    parentCategoryId?: string | null;
}

export interface Category {
    id: string;
    name: string;
    type: CategoryType;
    parentCategoryId: string | null;
    isArchived: boolean;
    children: Category[];
}

export enum CategoryType {
    Income = 1,
    Expense = 2
}

export const categoryTypeLabels: Record<CategoryType, string> = {
    [CategoryType.Income]: "Доход",
    [CategoryType.Expense]: "Расход"
};